web/components/PetPostModal/PetPostModal.tsx

```diff
@@ -323,18 +346,90 @@ const PetPostModal: React.FC<{
           paddingX={12}
           height={"inherit"}
         >
-          <Button
-            h={8}
-            w="fit-content"
-            bgColor="tag-primary-bg"
-            color="text-primary"
-            marginLeft={10}
-            _hover={{ bgColor: "tag-primary-bg" }}
-            leftIcon={<ArrowBackIcon />}
-            onClick={onClose}
-          >
-            Back to feed
-          </Button>
+          <Stack direction="row" justifyContent="space-between">
+            <Button
+              h={8}
+              w="fit-content"
+              bgColor="tag-primary-bg"
+              color="text-primary"
+              marginLeft={10}
+              _hover={{ bgColor: "tag-primary-bg" }}
+              leftIcon={<ArrowBackIcon />}
+              onClick={onClose}
+              id="backToFeedButton"
+            >
+              Back to feed
+            </Button>
+            {(role === Role.Admin || role === Role.ContentCreator) && (
+              <Button
+                h={8}
+                backgroundColor="white"
+                onClick={onDeleteConfirmationOpen}
+                _hover={{}}
+              >
+                <DeleteIcon marginRight="5px" color="#7D7E82" />
+                <Text textDecoration="underline" color="#7D7E82">
+                  Delete
+                </Text>
+              </Button>
```

Comment:
Use the `leftIcon` property built into Chakra - take a look at https://chakra-ui.com/docs/components/button#button-with-icon and